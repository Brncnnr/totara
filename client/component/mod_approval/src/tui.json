{
  "component": "mod_approval",
  "vendor": "totara",
  "dependencies": [
    "tui_xstate",
    "totara_engage",
    "totara_comment",
    "totara_notification",
    "editor_weka"
  ]
}
