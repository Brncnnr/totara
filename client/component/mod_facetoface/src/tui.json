{
  "component": "mod_facetoface",
  "vendor": "totara"
}
