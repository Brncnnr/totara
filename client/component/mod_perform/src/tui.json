{
  "component": "mod_perform",
  "vendor": "totara",
  "dependencies": [
    "totara_notification"
  ]
}
