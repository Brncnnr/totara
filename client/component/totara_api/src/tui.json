{
  "component": "totara_api",
  "vendor": "totara"
}