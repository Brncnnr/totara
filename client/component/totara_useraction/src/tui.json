{
  "component": "totara_useraction",
  "vendor": "totara"
}
