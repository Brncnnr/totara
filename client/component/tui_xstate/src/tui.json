{
  "component": "tui_xstate",
  "vendor": "totara",
  "entry": "./js/internal/entry.js"
}
