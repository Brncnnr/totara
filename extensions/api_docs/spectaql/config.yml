spectaql:
  themeDir: ./themes/totara
  errorOnInterpolationReferenceNotFound: false

introspection:
  removeTrailingPeriodFromDescriptions: false

info:
  title: API documentation
  x-hideWelcome: true
  x-introItems:
    - title: Endpoint
      file: ./guides/endpoint.md
    - title: Authentication
      file: ./guides/authentication.md
    - title: Request format
      file: ./guides/request_format.md
    - title: API schema
      file: ./guides/schema.md

servers:
  - url: https://YOUR-SITE-URL/api/graphql.php
    description: Production
    production: true

