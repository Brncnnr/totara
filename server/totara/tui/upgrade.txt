This file describes API changes in /tui/ information provided here is intended
especially for developers.

=== 16.0 ===

* Added a check to bail out early instead of hanging when compiling large non-ASCII SCSS files under PHP < 7.4, and added the pre_php74_allow_unperformant_css plugin setting to allow switching this off and attempting to compile anyway.

=== 15.0 ===

* Changed type of theme parameter from param_alphaext! to param_component! for query totara_tui_themes_with_variables.

=== 14.0 ===

* tenantid renamed to tenant_id in totara_tui\controllers\theme_settings