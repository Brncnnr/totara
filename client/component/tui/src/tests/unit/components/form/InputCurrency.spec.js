/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2021 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Tatsuhiro Kirihara <tatsuhiro.kirihara@totaralearning.com>
 * @module tui
 */

import { shallowMount } from '@vue/test-utils';
import component from 'tui/components/form/InputCurrency.vue';
import { config } from 'tui/config';

// Node 13 and below only supports English locales
config.locale.language = 'en';
config.locale.country = 'NZ';

let wrapper;
const inputEventFunc = jest.fn();
const submitEventFunc = jest.fn();

describe('InputCurrency', () => {
  beforeAll(() => {
    wrapper = shallowMount(component, {
      propsData: {
        ariaDescribedby: '#ariaDescription',
        ariaLabel: 'aria label',
        ariaLabelledby: '#ariaLabel',
        autocomplete: 'on',
        autofocus: true,
        disabled: true,
        id: 'inputid',
        max: 100,
        min: 0,
        name: 'currencyinput',
        pattern: '[0-9\\.]',
        placeholder: 'gimme money',
        readonly: true,
        required: true,
        step: 20,
        value: 123456.789,
        currency: 'USD',
      },
      listeners: {
        input: inputEventFunc,
        submit: submitEventFunc,
      },
    });
  });

  it('ariaDescribedby can be set', () => {
    let propValue = wrapper.find('#inputid').props().ariaDescribedby;
    expect(propValue).toEqual('#ariaDescription');
  });

  it('ariaLabel can be set', () => {
    let propValue = wrapper.find('#inputid').props().ariaLabel;
    expect(propValue).toEqual('aria label');
  });

  it('ariaLabelledby can be set', () => {
    let propValue = wrapper.find('#inputid').props().ariaLabelledby;
    expect(propValue).toEqual('#ariaLabel');
  });

  it('autocomplete can be set', () => {
    let propValue = wrapper.find('#inputid').props().autocomplete;
    expect(propValue).toEqual('on');
  });

  it('autofocus can be set', () => {
    let propValue = wrapper.find('#inputid').props().autofocus;
    expect(propValue).toBeTruthy();
  });

  it('disabled can be set', () => {
    let propValue = wrapper.find('#inputid').props().disabled;
    expect(propValue).toBeTruthy();
  });

  it('max can be set', () => {
    let propValue = wrapper.find('#inputid').props().max;
    expect(propValue).toEqual(100);
  });

  it('minlength can be set', () => {
    let propValue = wrapper.find('#inputid').props().min;
    expect(propValue).toEqual(0);
  });

  it('name can be set', () => {
    let propValue = wrapper.find('#inputid').props().name;
    expect(propValue).toEqual('currencyinput');
  });

  it('pattern can be set', () => {
    let propValue = wrapper.find('#inputid').props().pattern;
    expect(propValue).toEqual('[0-9\\.]');
  });

  it('placeholder can be set', () => {
    let propValue = wrapper.find('#inputid').props().placeholder;
    expect(propValue).toEqual('gimme money');
  });

  it('readonly can be set', () => {
    let propValue = wrapper.find('#inputid').props().readonly;
    expect(propValue).toBeTruthy();
  });

  it('required can be set', () => {
    let propValue = wrapper.find('#inputid').props().required;
    expect(propValue).toBeTruthy();
  });

  it('step can be set', () => {
    let propValue = wrapper.find('#inputid').props().step;
    expect(propValue).toEqual(20);
  });

  it('Value can be set', () => {
    let propValue = wrapper.find('#inputid').props().value;
    expect(propValue).toEqual(123456.789);
  });

  it('Input event can be triggered', () => {
    let input = wrapper.find('#inputid');
    input.vm.$emit('input', 'some value');
    expect(inputEventFunc).toHaveBeenCalled();
  });

  it('Submit event can be triggered', () => {
    let input = wrapper.find('#inputid');
    input.vm.$emit('submit');
    expect(submitEventFunc).toHaveBeenCalled();
  });

  it('Checks snapshot has not changed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Checks snapshot has not changed with default step', () => {
    const propsData = { currency: 'GBP' };
    wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Checks snapshot has not changed with custom currency', () => {
    const propsData = {
      currency: { currency: 'JPY', symbol: 'å††', side: 'end', fractions: 1 },
    };
    wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });
});
