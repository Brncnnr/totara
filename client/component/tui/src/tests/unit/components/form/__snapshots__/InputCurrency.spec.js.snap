// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCurrency Checks snapshot has not changed 1`] = `
<div
  class="tui-formInputCurrency"
>
  <span
    class="tui-formInputCurrency__symbol"
  >
    
    $
  
  </span>
   
  <input-stub
    ariadescribedby="#ariaDescription"
    arialabel="aria label"
    arialabelledby="#ariaLabel"
    autocomplete="on"
    autofocus="true"
    charlength="full"
    disabled="true"
    id="inputid"
    max="100"
    min="0"
    name="currencyinput"
    pattern="[0-9\\\\.]"
    placeholder="gimme money"
    readonly="true"
    required="true"
    step="20"
    styleclass="[object Object]"
    type="number"
    value="123456.789"
  />
   
  <!---->
</div>
`;

exports[`InputCurrency Checks snapshot has not changed with custom currency 1`] = `
<div
  class="tui-formInputCurrency"
>
  <!---->
   
  <input-stub
    charlength="full"
    step="0.1"
    styleclass="[object Object]"
    type="number"
  />
   
  <span
    class="tui-formInputCurrency__symbol"
  >
    
    円
  
  </span>
</div>
`;

exports[`InputCurrency Checks snapshot has not changed with default step 1`] = `
<div
  class="tui-formInputCurrency"
>
  <span
    class="tui-formInputCurrency__symbol"
  >
    
    £
  
  </span>
   
  <input-stub
    charlength="full"
    step="0.01"
    styleclass="[object Object]"
    type="number"
  />
   
  <!---->
</div>
`;
